activity_browser/settings.py

```diff
@@ -77,46 +74,55 @@ def __init__(self, filename: str):
 
     @staticmethod
-    def update_old_settings(directory: str, filename: str) -> None:
-        """Recycling code to enable backward compatibility: This function is only required for compatibility
-        with the old settings file and can be removed in a future release
+    def _update_old_settings(directory: Path, filename: str) -> None:
+        """Update old settings, backwards compatibility method.
+
+        This function is only required for compatibility with the old settings file
+        and can be removed in a future release.
+
+        Parameters
+        ----------
+        directory : Path
+            Path to the directory containing the settings.
+        filename : str
+            Name of the settings file.
         """
-        file = os.path.join(directory, filename)
-        if not os.path.exists(file):
+        new_file = directory.joinpath(filename)
+        if not new_file.exists():
             package_dir = Path(__file__).resolve().parents[1]
-            old_settings = os.path.join(package_dir, "ABsettings.json")
-            if os.path.exists(old_settings):
-                shutil.copyfile(old_settings, file)
-        if os.path.isfile(file):
-            with open(file, "r") as current:
+            old_settings = package_dir.joinpath("ABsettings.json")
+            if old_settings.exists():
+                shutil.copyfile(old_settings, new_file)
+        if new_file.is_file():
+            with new_file.open() as current:
                 current_settings = json.load(current)
             if "current_bw_dir" not in current_settings:
                 new_settings_content = {
                     "current_bw_dir": current_settings["custom_bw_dir"],
                     "custom_bw_dirs": [current_settings["custom_bw_dir"]],
                     "startup_project": current_settings["startup_project"],
                 }
-                with open(file, "w") as new_file:
+                with new_file.open("w") as new_file:
                     json.dump(new_settings_content, new_file)
 
     @classmethod
     def get_default_settings(cls) -> dict:
-        """Using methods from the commontasks file to set default settings"""
+        """Using methods from the commontasks file to set default settings."""
         return {
-            "current_bw_dir": cls.get_default_directory(),
-            "custom_bw_dirs": [cls.get_default_directory()],
+            "current_bw_dir": str(bw2data.projects._base_data_dir),
+            "custom_bw_dirs": [str(bw2data.projects._base_data_dir)],
             "startup_project": cls.get_default_project_name(),
         }
 
     @property
     def custom_bw_dir(self) -> str:
-        """Returns the custom brightway directory, or the default"""
-        return self.settings.get("custom_bw_dirs", self.get_default_directory())
+        """Returns the custom brightway directory, or the default."""
+        return self.settings.get("custom_bw_dirs") or str(bw2data.projects._base_data_dir)
```

Comment:
This is an important point, and I think we should document it more clearly. AFAICT there is a "default" directory set at startup, and this is what we can revert to at anytime. We can also change the data directory, but the only history of these changes that is made is that we remember the "default" directory.